brc.md

```diff
@@ -0,0 +1,92 @@
+
+## Accessing BRC via SSH
+
+To access BRC, use the following SSH command:
+```
+ssh yourusername@hpc.brc.berkeley.edu
+```
+
+- At the `Password:` prompt, enter your token PIN followed immediately (without spaces) by the 6-digit one-time password from the Google Authenticator app.
+  - **Example**: If your password is `ilovebrc` and the one-time code is `123456`, you would type `ilovebrc123456`.
+
+## Transferring Singularity Containers
+
+1. **Download Required Containers**
+   - Libraries container: [Download Link](https://drive.google.com/file/d/1syZVEb0l5d75q7NDaQRyS7FJQCqSxlDq/view?usp=sharing)
+   - CUDA container: [Download Link](https://drive.google.com/file/d/1gF3-sQoQewP6Dhv1jVxbYvqRbrLT2idS/view?usp=sharing)
+   - **Note**: Store files other than code repositories and some models in the scratch directory (`/global/scratch/users/<username>/`).
+
+2. **Upload Containers to Scratch Directory**
+   - Use the `scp` command to transfer the downloaded containers to your scratch directory on BRC:
+     ```
+     scp -r ~/Downloads/overlay-50G-10M.ext3.gz <username>@dtn.brc.berkeley.edu:/global/scratch/users/<username>/singularity/
+     scp -r ~/Downloads/cuda11.5-cudnn8-devel-ubuntu18.04.sif <username>@dtn.brc.berkeley.edu:/global/scratch/users/<username>/singularity/
+     ```
+   - Unzip the file in the target directory:
+     ```
+     gunzip /global/scratch/users/<username>/singularity/overlay-50G-10M.ext3.gz
+     ```
+
+## Setting Up the Environment
+
+1. **Obtain GPU Access**
+   - Use the `srun` command to allocate a GPU for setting up your conda environment:
+     ```
+     srun --account=fc_ocow --partition=savio4_gpu --cpus-per-task=4 --gres=gpu:A5000:1 --qos=a5k_gpu4_normal -t 3:00:00 --pty /bin/bash
+     ```
+   - For different GPU/partition options, check available associations with:
+     ```
+     sacctmgr show associations where account=fc_ocow format=Account,Partition,qos
+     ```
+
+2. **Start Singularity Container**
+   - Set the scratch directory variable and execute the singularity container:
+     ```
+     export SCRATCH='/global/scratch/users/<username>'
+     singularity exec --userns --fakeroot --nv -B /usr/lib64 -B /var/lib/dcv-gl --overlay $SCRATCH/singularity/overlay-50G-10M.ext3:rw $SCRATCH/singularity/cuda11.5-cudnn8-devel-ubuntu18.04.sif /bin/bash
+     ```
+
+3. **Install Conda Environment**
+   - Navigate to `/ext3` and install Miniconda:
+     ```
+     cd /ext3
+     mkdir miniconda3
+     wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda3/miniconda.sh
+     bash miniconda3/miniconda.sh -b -u -p miniconda3
+     rm -rf miniconda3/miniconda.sh
+     miniconda3/bin/conda init bash
+     ```
+   - Use `conda env export > environment.yml` on your machine, adjust the prefix in `environment.yml`, and then create the conda environment on BRC with `conda env create -f environment.yml`.
```

Comment:
This is a BRC in general explainer, since the process is basically identical and for now this is probably the easiest way to share the file with everyone.